prov/sockets/include/sock.h

```diff
@@ -870,7 +870,12 @@ struct sock_conn_hdr {
 	uint8_t reserved[3];
 	uint16_t port;
 	uint16_t cm_data_sz;
-	char cm_data[0];
+	/* Dear developer, this value is commented due to
+	 * issues with building on non-GCC compilers
+	 * (MS Visual Studio Compiler). Please, do not
+	 * uncomment it until you really need this value.
+	 * Thank you. */
+	/*char cm_data[0];*/
```

Comment:
Let's either replace cm_data[0] with this comment:
```
/* cm data follows cm_data_sz */
```
or define it as cm_data[], which should also work.  The field is not being accessed, but the data area is.  The location is just assumed to follow sock_conn_hdr.